notebooks/examples/python/Ship2ShipTransfers/00. Data Prep.py

```diff
@@ -0,0 +1,200 @@
+  Length int, 
+  Width int, 
+  Draft double, 
+  Cargo int, 
+  TranscieverClass string
+"""
+
+AIS_df = (
+    spark.read.option("badRecordsPath", "/tmp/ais_invalid")  # Quarantine bad records
+    .csv("/tmp/vessels/2018", header=True, schema=schema)
+    .filter("VesselType = 70")  # Only select cargos
+    .filter("Status IS NOT NULL")
+)
+display(AIS_df)
+
+# COMMAND ----------
+
+# MAGIC %sql
+# MAGIC CREATE DATABASE IF NOT EXISTS ship2ship
+
+# COMMAND ----------
+
+(AIS_df.write.format("delta").mode("overwrite").saveAsTable("ship2ship.AIS"))
+
+# COMMAND ----------
+
+# MAGIC %md ## Continental US Data
+
+# COMMAND ----------
+
+from pyspark.sql.functions import *
+
+one_metre = 0.00001 - 0.000001
+tolerance = 500 * one_metre
+
+# COMMAND ----------
+
+user_name = (
+    dbutils.notebook.entry_point.getDbutils().notebook().getContext().userName().get()
+)
+
+raw_path = f"dbfs:/tmp/ship2ship/{user_name}"
+raw_us_outline_path = f"{raw_path}/us_outline"
+
+print(f"The raw data is stored in {raw_path}")
+
+# COMMAND ----------
+
+import requests
+import pathlib
+
+us_outline_url = (
+    "https://eric.clst.org/assets/wiki/uploads/Stuff/gz_2010_us_outline_500k.json"
+)
+
+# The DBFS file system is mounted under /dbfs/ directory on Databricks cluster nodes
+
+local_us_outline_path = pathlib.Path(raw_us_outline_path.replace("dbfs:/", "/dbfs/"))
+local_us_outline_path.mkdir(parents=True, exist_ok=True)
+
+req = requests.get(us_outline_url)
+with open(local_us_outline_path / f"us_outline.geojson", "wb") as f:
+    f.write(req.content)
+
+# COMMAND ----------
+
+outlines = (
+    spark.read.format("json")
+    .option("multiline", "true")
+    .load(raw_us_outline_path)
+    .select("type", explode(col("features")).alias("feature"))
+    .select(
+        "type",
+        col("feature.properties").alias("properties"),
+        to_json(col("feature.geometry")).alias("json_geometry"),
+    )
+    .withColumn("geometry", mos.st_aswkt(mos.st_geomfromgeojson("json_geometry")))
+    .withColumn("coastal_area", mos.st_buffer("geometry", lit(tolerance)))
+    .where("properties.type == 'COASTAL'")
+    .drop("json_geometry")
+)
+display(outlines)
+
+# COMMAND ----------
+
+# MAGIC %%mosaic_kepler
+# MAGIC outlines "coastal_area" "geometry" 100
+
+# COMMAND ----------
+
+coastal_h3 = (
+    outlines.withColumn("geometry", mos.st_buffer("geometry", lit(tolerance)))
+    .select(mos.mosaic_explode("geometry", lit(9)).alias("h3"))
+    .select(col("h3.index_id").alias("h3"))
+)
+
+coastal_h3.createOrReplaceTempView("coastal")
+
+display(coastal_h3)
+
+# COMMAND ----------
+
+# MAGIC %%mosaic_kepler
+# MAGIC "coastal" "h3" "h3" 100000
+
+# COMMAND ----------
+
+(coastal_h3.write.mode("overwrite").format("delta").saveAsTable("ship2ship.coastal_h3"))
+
+# COMMAND ----------
+
+# MAGIC %md ## Harbours
+# MAGIC
+# MAGIC This data can be obtained from [here](https://data-usdot.opendata.arcgis.com/datasets/usdot::ports-major/about), and loaded accordingly.
```

Comment:
Could you please add the code to load this data as well? Ideally this example would have to run as-is on any workspace without manual steps.

Reply:
That was certainly my hope, but I was not able to find a programmatic way of accessing the data. 
Are you able to see one? 

https://data-usdot.opendata.arcgis.com/datasets/usdot::ports-major/about

Reply:
The latest version should pull it directly form the source.